### ISSUE TITLE
EntropySGD: gradient updates for Langevin dynamics

### ISSUE THREAD

**Issue by maxleverage** [fchollet/keras#6631]
Hi, 

I've been attempting to implement the EntropySGD in this paper (https://arxiv.org/pdf/1611.01838.pdf). However the inner loop of the optimizer requires L gradient updates to estimate the exponential decayed mus which are used to update the parameters on one batch of data. It looks like  'get_updates' is called from training.py. My question is:

1. Is there a way to update the network params, compute a new loss and get new gradients on the same batch of data in the optimizer class to run the L iterations for the Langevin dynamics?
2. If not, can this be implemented in other ways or can this be added as a future feature?

Thanks

**Comment by maxleverage** [comment]
Being implemented https://github.com/fchollet/keras/issues/6563
